stdlib/public/runtime/Casting.cpp

```diff
@@ -949,7 +966,7 @@ static bool _dynamicCastToExistential(OpaqueValue *dest,
     auto object = *(reinterpret_cast<HeapObject**>(srcDynamicValue));
     destExistential->Value = object;
     if (!canConsumeDynamicValue || !(flags & DynamicCastFlags::TakeOnSuccess)) {
-      swift_retain(object);
+      swift_unknownRetain(object);
```

Comment:
(I mean we don't seem to ultimately end up with 'take'ing address cast in such a context - at least up to unreported/screen bugs ...)